satellite/rewards/display.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package rewards
+
+// OrganizedOffers contains a list of offers organized by status.
+type OrganizedOffers struct {
+	Active  Offer
+	Default Offer
+	Done    Offers
+}
+
+// OpenSourcePartner contains all data for an Open Source Partner.
+type OpenSourcePartner struct {
+	PartnerInfo
+	PartnerOffers OrganizedOffers
+}
+
+// PartnerSet contains a list of Open Source Partners.
+type PartnerSet []OpenSourcePartner
+
+// OfferSet provides a separation of marketing offers by type.
+type OfferSet struct {
+	ReferralOffers OrganizedOffers
+	FreeCredits    OrganizedOffers
+	PartnerTables  PartnerSet
+}
+
+type referralInfo struct {
+	UserID    string
+	PartnerID string
+}
+
+// OrganizeOffersByStatus organizes offers by OfferStatus.
+func (offers Offers) OrganizeOffersByStatus() OrganizedOffers {
+	var oo OrganizedOffers
+
+	for _, offer := range offers {
+		switch offer.Status {
+		case Active:
```

Comment:
do we need to worry about whether there might be multiple offers where `offer.Status` is `Active` or `Default`?

Reply:
There should be only one Active or Default offer at a given time